lib/exceptions.h

```diff
@@ -54,27 +56,47 @@ class CompilerBug final : public P4CExceptionBase {
  public:
     template <typename... T>
     CompilerBug(const char* format, T... args)
-            : P4CExceptionBase(format, args...)
-    { message = cstring(ANSI_RED) + "Compiler Bug" + ANSI_CLR + ":\n" + message; }
+            : P4CExceptionBase(format, args...) {
+        // Check if output is redirected and if so, then don't color text so that
+        // escape characters are not present
+        const bool isCerrRedirected = ttyname(fileno(stderr)) == nullptr;  // NOLINT(runtime/threadsafe_fn)
```

Comment:
Please move this code into a helper function, e.g., `colorize`.
You can also probably cache the result of the check.

Reply:
Sure thing, I moved it to its own function.